design/baremetal-operator/external-introspection.md

```diff
@@ -0,0 +1,116 @@
+# Support external introspection
+
+## Status
+
+implementable
+
+## Summary
+
+A declarative API is proposed to disable inspection of a BareMetalHost
+and optionally allow external sources of inspection data to update the
+hardware status data.
+
+## Motivation
+
+Related to the work to enable booting of a
+[LiveImage](https://github.com/metal3-io/metal3-docs/pull/150),
+there is the requirement to optionally disable inspection on initial
+registration of a BareMetalHost (so that the live image can boot
+more quickly, ref user stories below).
+
+### Goals
+
+- A declarative API to disable inspection on BMH registration
+- Provide an interface to update hardware status data when inspection is
+  performed by some means other than the regular BMH flow at an arbitrary time
+  after BMH creation.
+- An interface that's consistent with existing proposals, in particular the
+  [re-inspection interface](https://github.com/metal3-io/metal3-docs/blob/master/design/baremetal-operator/re-introspection-interface.md)
+
+### Non-Goals
+
+- There are no plans for any mechanism to trigger or consume data from any out
+  of band inspection process, other than providing data via a status annotation.
+
+## Proposal
+
+### Disable inspection proposal
+
+To align with the re-inspection API proposal, the `inspect.metal3.io`
+annotation will be reused, but with an optional additional suffix, similar
+to that described for the [reboot API](https://github.com/metal3-io/metal3-docs/blob/master/design/baremetal-operator/reboot-interface.md)
+
+The optional `inspect.metal3.io/disable` annotation will be used to describe
+the situation where we wish to disable the default inspection behavior.
+
+When the BMO finds this annotation, it will skip performing inspection
+during the
+[Inspecting state](https://github.com/metal3-io/baremetal-operator/blob/master/docs/BaremetalHost_ProvisioningState.png)
+
+In this case (similar to the reboot interface), when the suffixed form of
+`inspect.metal3.io/disable` is found, it will never be removed by the BMO.
+
+If both inspect.metal3.io and inspect.metal3.io/disable forms are specified,
+the `inspect.metal3.io/disable` form will take priority and the
+`inspect.metal3.io` annotation will be ignored.
+
+### Hardware status update proposal
+
+In the current implementation, when `baremetalhost.metal3.io/status` is
+provided, it can set any status field, and thus is only evaluated on the
+very first reconcile (primarily to support externally provisioned hosts,
+where we collect the inspection data prior to creating the BMH resources).
+
+In the case where metal3 is booting a live-image that contains code that
+can collect hardware details, it's desirable to have a way to update the
+hardware status after the image is booted.
+
+To enable this safely, we can add a new `baremetalhost.metal3.io/status/hardware`
```

Comment:
I don't think annotation names can contain multiple `/` characters (https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set). If we don't go with @zaneb's proposal above, this could be `baremetalhost.metal3.io/hardware` or `baremetalhost.metal3.io/status-hardware`.